src/driver/amdxdna/aie2_pci.c

```diff
@@ -1005,43 +1005,70 @@ static int aie2_query_ctx_status(struct amdxdna_client *client,
 static int aie2_query_telemetry(struct amdxdna_client *client,
 				struct amdxdna_drm_get_info *args)
 {
+	struct amdxdna_drm_query_telemetry_header __user *header;
+	struct amdxdna_drm_query_telemetry_header *tmp;
 	struct amdxdna_dev *xdna = client->xdna;
-	struct amdxdna_dev_hdl *ndev;
+	size_t aligned_sz, offset, map_size;
 	struct aie_version ver;
 	dma_addr_t dma_addr;
-	size_t aligned_sz;
 	void *buff;
+	int ret, i;
 	u32 type;
-	int ret;
 
 	if (!access_ok(u64_to_user_ptr(args->buffer), args->buffer_size)) {
 		XDNA_ERR(xdna, "Failed to access buffer size %d", args->buffer_size);
 		return -EFAULT;
 	}
 
-	if (copy_from_user(&type, u64_to_user_ptr(args->buffer), sizeof(type))) {
+	header = (struct amdxdna_drm_query_telemetry_header *)u64_to_user_ptr(args->buffer);
+	if (copy_from_user(&type, &header->type, sizeof(type))) {
```

Comment:
"header" is a user pointer. "header->type" directly dereference it. This will work without issue on x86. But this is not allow in upstreaming point of view.